app/src/main/java/de/madem/homium/utilities/extensions/AndroidCoreExtensions.kt

```diff
@@ -266,4 +267,11 @@ fun Fragment?.whenSearchViewHandler(action: (SearchViewHandler) -> Unit){
     s.notNull {
         action.invoke(it)
     }
+}
+
+fun <T> Flow<T>.collectFor(viewLifecycleOwner: LifecycleOwner, onEmit: suspend (T) -> Unit) {
```

Comment:
Suggestion: Maybe we should call it `onCollect` instead. :D

Reply:
yeah probably a better name